lib/net.c

```diff
@@ -696,36 +696,39 @@ init_iface(struct gatekeeper_if *iface)
 	}
 
 	/* Initialize bonded port, if needed. */
-	if (iface->num_ports == 1) {
+	if (iface->num_ports == 1)
 		iface->id = iface->ports[0];
-		return 0;
-	}
-
-	/* TODO Also allow LACP to be used. */
-	ret = rte_eth_bond_create(iface->name, BONDING_MODE_ROUND_ROBIN, 0);
-	if (ret < 0) {
-		RTE_LOG(ERR, PORT, "Failed to create bonded port (err=%d)!\n",
-			ret);
-		goto close_partial;
-	}
+	else {
+		/* TODO Also allow LACP to be used. */
+		ret = rte_eth_bond_create(iface->name,
+			BONDING_MODE_ROUND_ROBIN, 0);
+		if (ret < 0) {
+			RTE_LOG(ERR, PORT,
+				"Failed to create bonded port (err=%d)!\n",
+				ret);
+			goto close_partial;
+		}
 
-	iface->id = (uint8_t)ret;
+		iface->id = (uint8_t)ret;
+
+		for (i = 0; i < iface->num_ports; i++) {
```

Comment:
I think the diff is a little hard to read, but the loop is already in the `else` clause here.